climada/hazard/tc_tracks.py

```diff
@@ -1275,25 +1279,22 @@ def write_hdf5(self, file_name, complevel=5):
             Specifies a compression level (0-9) for the zlib compression of the data.
             A value of 0 or None disables compression. Default: 5
         """
-        # initialise the data set with the size information
-        xr.Dataset(attrs={'size': self.size}).to_netcdf(file_name, mode="w", format="NetCDF4")
-
-        LOGGER.info('Writing %d tracks to %s', self.size, file_name)
-        last_perc = 0
         for i, track in enumerate(self.data):
-            perc = 100 * i / self.size
-            if perc - last_perc >= 10:
-                LOGGER.info("Progress: %d%%", perc)
-                last_perc = perc
-
             # change dtype from bool to int to be NetCDF4-compliant
             track.attrs['orig_event_flag'] = int(track.attrs['orig_event_flag'])
-            encoding = {var: dict(zlib=True, complevel=complevel) for var in track.data_vars}
-            track.to_netcdf(file_name, mode="a", group=f"track{i}", encoding=encoding)
-            track.attrs['orig_event_flag'] = bool(track.attrs['orig_event_flag'])
-
-        if last_perc != 100:
-            LOGGER.info("Progress: 100%")
+        try:
+            encoding = {
+                f'track{i}': {var: dict(zlib=True, complevel=complevel) for var in track.data_vars}
+                for i, track in enumerate(self.data)
+            }
+            ds_dict = {f'track{i}': track for i, track in enumerate(self.data)}
+            ds_dict[""] = xr.Dataset(attrs={'size': self.size})
+
+            LOGGER.info('Writing %d tracks to %s', self.size, file_name)
+            _xr_to_netcdf_multi(file_name, ds_dict, encoding=encoding)
+        finally:
+            for i, track in enumerate(self.data):
+                track.attrs['orig_event_flag'] = bool(track.attrs['orig_event_flag'])
```

Comment:
Why do you change the track object after saving to file?

Reply:
This reverses the change that has been applied before saving to the file, see the comments in fb1655b8